### PR TITLE
[CDAP-15425] Add Textbox widget react component and share with Angular side

### PR DESCRIPTION
**Pull request by yukiej**
JIRA: https://issues.cask.co/browse/CDAP-15425
BUILD: https://builds.cask.co/browse/CDAP-UDUT314-5

### CODE REVIEW

File: cdap-ui/app/directives/macro-widget-toggle/macro-widget-toggle.html [hunk 1]
```diff
@@ -97,11 +99,16 @@
         }"
       >
         <!-- If its macro enabled show a textbox -->
-        <input
+        <text-box
+          id="macro-input-{{MacroWidget.field.name}}"
+          value="MacroWidget.node.plugin.properties[MacroWidget.field.name]"
+          on-change="MacroWidget.onChange"
+        ></text-box>
+        <!-- <input
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/macro-widget-toggle/macro-widget-toggle.html]: nit: Can we remove the commented out code. We can always go back in git history if we need that snippet.

File: cdap-ui/app/cdap/components/FormInputs/TextBox/index.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import React from 'react';
+import TextField from '@material-ui/core/TextField';
+import PropTypes from 'prop-types';
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import ThemeWrapper from 'components/ThemeWrapper';
+
+const styles = (theme) => {
+  return {
+    ':hover': {
+      borderColor: '#cccccc',
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/FormInputs/TextBox/index.tsx]: nit: The color should be used from theme. (`theme.palette.grey['50']`) + same for all color references in this style.

**Review by ajainarayanan — APPROVED** [review]
Minor comment. Otherwise 👍 LGTM

File: cdap-ui/app/cdap/components/FormInputs/TextBox/index.tsx (hunk 2 above)
Comment (yukiej) [reply cdap-ui/app/cdap/components/FormInputs/TextBox/index.tsx]: Thanks! Grey['50'] is pretty dark - it looks more like grey['400']?

Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/FormInputs/TextBox/index.tsx]: SG!

**Review by ajainarayanan — APPROVED** [review]
👍 LGTM.

nit: While squashing I guess you have included all the intermediate commit messages. It would be helpful if you could write only the high level changes or skip the commit message entirely. Its fine for now but may be next time

**Comment by yukiej** [comment]
Latest green build: https://builds.cask.co/browse/CDAP-UDUT314-6

**Pull request merged**